### PR TITLE
refactor(parsers): rewrote parsers using ts-parsec and added deparser functionality

### PR DESCRIPTION
**Pull request by endes0**
This PR refactors all the 3 parsers present in the code, replacing the old Lex dependency with ts-parsec. While it maintains retrocompatibility with the old file formats, this new parsers are more flexible(EX: The first line, with indicate the number of instructions is not necessary anymore).
This Pr also implements a new functionality, deparsing, allowing to download content and code from the interface. 
Finally, it fixes some bugs. A notorious example is that test in subdirs were currently not being executed with` npm run test`

### CODE REVIEW

File: src/core/Common/Code.ts
Comment (oxcabe) [inline src/core/Common/Code.ts:1]: Looks way better now!

File: src/core/Common/InstructionFormats.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+export enum Formats {
+    TwoGeneralRegisters = 0,        // OP R1, R2, R3
+    TwoFloatingRegisters,           // OP F1, F2, F3
+    GeneralRegisterAndInmediate,    // OP R1, R2, #X
+    GeneralLoadStore,               // OP R1, X(R2)
+    FloatingLoadStore,              // OP F1, X(R2)
+    Jump,                           // OP R1, R2, label
+    NooP                            // NOP
```
Comment (oxcabe) [inline src/core/Common/InstructionFormats.ts:11]: Refactor this to `Noop`

File: src/i18n/locales/en/common.json [hunk 2]
```diff
@@ -2,7 +2,10 @@
     "fileBar": {
         "file": {
             "name": "File",
-            "load": "Load"
+            "load": "Load",
+            "download_memory": "Download Memory(JSON)",
```
Comment (oxcabe) [inline src/i18n/locales/en/common.json:6]: Use camelCase for any JSON fields you create or modify

File: src/core/Common/Code.ts [hunk 3]
```diff
@@ -297,7 +183,90 @@ export class Code {
...
+        return result;
+    }
+
+    public getBasicBlockInstruction(basicBlockIndex: number) {
+        if (basicBlockIndex > this._numberOfBlocks) {
+            //TODO: throw exception
+            return -1;
+        }
+        let actual: BasicBlock = this._basicBlocks;
+        // WHaat why we are reinventing the wheel using a linked list?
```
Comment (oxcabe) [inline src/core/Common/Code.ts:264]: Avoid writing comments like this one 😅

**Review by oxcabe — CHANGES_REQUESTED** [review]
Good implementation. There is still a lot of room for improvement, though! Try to avoid writing all these `// TODO: ...` comments. Next time, instead, do a follow-up issue that references the working PR with the future implementations or refactor you find necessary.

It'd also be best practice to run a code formatter tool (such as [Prettier](https://prettier.io/)) that tidies the files you are about to commit. I wouldn't bother doing it for this PR, but could be something mandatory if we decide to integrate something like [Husky](https://github.com/typicode/husky) after the Vite tooling refactor issue https://github.com/etsiiull/SIMDE/issues/66 is fulfilled.

I'd also recommend you to use a [LSP-based](https://github.com/typescript-language-server/typescript-language-server) linter for TypeScript if you aren't already using one. This is something that will probably be configured in the future directly into the TypeScript building process, or through the Husky integration.

Submit the requested changes and we're good to go. Thank you for your contribution!

**Comment by oxcabe** [comment]
Oh, and one last thing. Rename the PR title to adhere to the [Conventional Commits v1.0.0 standard](https://www.conventionalcommits.org/en/v1.0.0/).

**Comment by endes0** [comment]
> Good implementation. There is still a lot of room for improvement, though! Try to avoid writing all these `// TODO: ...` comments. Next time, instead, do a follow-up issue that references the working PR with the future implementations or refactor you find necessary.


I have to admit that at the end I rushed this PR and I forgot to do the usual cleanup commit, fixing it now.

> 
> It'd also be best practice to run a code formatter tool (such as [Prettier](https://prettier.io/)) that tidies the files you are about to commit. I wouldn't bother doing it for this PR, but could be something mandatory if we decide to integrate something like [Husky](https://github.com/typicode/husky) after the Vite tooling refactor issue #66 is fulfilled.
> 
> I'd also recommend you to use a [LSP-based](https://github.com/typescript-language-server/typescript-language-server) linter for TypeScript if you aren't already using one. This is something that will probably be configured in the future directly into the TypeScript building process, or through the Husky integration.
> 
> Submit the requested changes and we're good to go. Thank you for your contribution!

Yes, I am using VSCode with I think have an LSP-based linter and a code formatter (which I also forgot to use) by default. Btw, I tought that the project had already a linter configured during build or at least there is an `lint` script and config(`tslint.json`) present.

**Review by oxcabe — APPROVED** [review]

**Pull request merged**